### ISSUE TITLE
Toolchain build 2.8.0 is broken (binutils-gdb issue)

### ISSUE THREAD

**Issue by mshabunin** [XUANTIE-RV/xuantie-gnu-toolchain#11]
When building toolchain per instruction an issue appears:
```
In file included from /xuantie-gnu-toolchain/riscv-gdb/gdb/arch/riscv.c:19:
/xuantie-gnu-toolchain/riscv-gdb/gdb/arch/riscv.h:103:2: error: #endif without #if
  103 | #endif /* ARCH_RISCV_H */
      |  ^~~~~
```
Mentioned file has broken preprocessor macro possibly due to merge issue. Problem can be fixed by editing that riscv.h file.

_Note:_ this ticket is a copy of https://github.com/T-head-Semi/binutils-gdb/issues/3 made for visibility

**Comment by JF002** [comment]
I ran into the same issue. I guess this error comes from incomplete [preprocessor instructions](https://github.com/T-head-Semi/binutils-gdb/blob/66d5fa3a5aea033a6822bdcaf85e31c48048280d/gdb/arch/riscv.h#L57) : there are `#else` and `#endif` bot no `if`.

Now the question is : which one of those equality operators should we keep? I removed the one between the `#else` and `#end` and it builds, but I don't know if it's correct.

**Comment by Cookfeces** [comment]
Duplicate to https://github.com/T-head-Semi/binutils-gdb/issues/3

**Comment by Cooper-Qu** [comment]
Duplicate to https://github.com/T-head-Semi/xuantie-gnu-toolchain/issues/11

**Comment by JF002** [comment]
I guess you meant duplicate to https://github.com/T-head-Semi/binutils-gdb/issues/3 ?

And thank you so much for the fix in https://github.com/T-head-Semi/binutils-gdb/commit/3f1c1bda96771d78453da163ef7aedca540d0f18!

**Comment by Cooper-Qu** [comment]
> I guess you meant duplicate to [T-head-Semi/binutils-gdb#3](https://github.com/T-head-Semi/binutils-gdb/issues/3) ?
> 
> And thank you so much for the fix in [T-head-Semi/binutils-gdb@3f1c1bd](https://github.com/T-head-Semi/binutils-gdb/commit/3f1c1bda96771d78453da163ef7aedca540d0f18)!

Thank you for correcting my comment.
